File: java-testng/src/test/java/com/testng/coffeemachine/ProjectTest.java
package com.testng.coffeemachine;

import org.testng.annotations.*;

public class ProjectTest {

    public Actionwords actionwords;

    @BeforeMethod
    public void setUp() {
        actionwords = new Actionwords();
    }
    public void simpleUse(String lang, String readyMessage) {
        // Given I start the coffee machine "<lang>"
        actionwords.iStartTheCoffeeMachine(lang);
        // When I take a coffee
        actionwords.iTakeACoffee();
        // Then coffee should be served
        actionwords.coffeeShouldBeServed();
    }

    @Test
    public void simpleUseEnglishUid858d270d51854e2b82ba5ddfe219fe42() {
        simpleUse("en", "Ready");
    }

    @Test
    public void simpleUseFrenchUid373898f68647419d80deea6c3ccfb894() {
        simpleUse("fr", "Pret");
    }



    @Test
    public void fullGroundsDoesNotBlockCoffeeUide06952a328e34918899bb408596cac6c() {
        // Given the coffee machine is started
        actionwords.theCoffeeMachineIsStarted();
        // When I take "29" coffees
        actionwords.iTakeCoffeeNumberCoffees(29);
        // Then message "Ready" should be displayed
        actionwords.messageMessageShouldBeDisplayed("Ready");
        // When I take a coffee
        actionwords.iTakeACoffee();
        // Then coffee should be served
        actionwords.coffeeShouldBeServed();
        // And message "Empty grounds" should be displayed
        actionwords.messageMessageShouldBeDisplayed("Empty grounds");
        // When I fill the water tank
        actionwords.iFillTheWaterTank();
        // And I fill the beans tank
        actionwords.iFillTheBeansTank();
        // And I take "20" coffees
        actionwords.iTakeCoffeeNumberCoffees(20);
        // Then coffee should be served
        actionwords.coffeeShouldBeServed();
        // And message "Empty grounds" should be displayed
        actionwords.messageMessageShouldBeDisplayed("Empty grounds");
    }

    @Test
    public void waterRunsAwayUid1cae567daeb34449b10c3457fbcf2661() {
        // Given the coffee machine is started
        actionwords.theCoffeeMachineIsStarted();
        // When fifty coffees have been taken without filling the tank
        actionwords.fiftyCoffeesHaveBeenTakenWithoutFillingTheTank();
        // Then message "Fill tank" should be displayed
        actionwords.messageMessageShouldBeDisplayed("Fill tank");
        // When I take a coffee
        actionwords.iTakeACoffee();
        // Then coffee should be served
        actionwords.coffeeShouldBeServed();
        // When I take "10" coffees
        actionwords.iTakeCoffeeNumberCoffees(10);
        // Then coffee should not be served
        actionwords.coffeeShouldNotBeServed();
        // And message "Fill tank" should be displayed
        actionwords.messageMessageShouldBeDisplayed("Fill tank");
        // When I fill the water tank
        actionwords.iFillTheWaterTank();
        // Then message "Ready" should be displayed
        actionwords.messageMessageShouldBeDisplayed("Ready");
    }

    @Test
    public void beansRunOutUid186d320e413b4ac697b47860ece64da9() {
        // Given the coffee machine is started
        actionwords.theCoffeeMachineIsStarted();
        // When thirty eight coffees are taken without filling beans
        actionwords.thirtyEightCoffeesAreTakenWithoutFillingBeans();
        // Then coffee should be served
        actionwords.coffeeShouldBeServed();
        // And message "Fill beans" should be displayed
        actionwords.messageMessageShouldBeDisplayed("Fill beans");
        // When I take "2" coffees
        actionwords.iTakeCoffeeNumberCoffees(2);
        // Then coffee should be served
        actionwords.coffeeShouldBeServed();
        // And message "Fill beans" should be displayed
        actionwords.messageMessageShouldBeDisplayed("Fill beans");
        // When I take a coffee
        actionwords.iTakeACoffee();
        // Then coffee should not be served
        actionwords.coffeeShouldNotBeServed();
    }

    @Test
    public void noMessagesAreDisplayedWhenMachineIsShutDownUid78707faf2f21450cb3acbb543192fdf4() {
        // Given the coffee machine is started
        actionwords.theCoffeeMachineIsStarted();
        // When I shutdown the coffee machine
        actionwords.iShutdownTheCoffeeMachine();
        // Then message "" should be displayed
        actionwords.messageMessageShouldBeDisplayed("");
    }
    public void messagesAreBasedOnLanguage(String lang, String readyMessage) {
        // When I start the coffee machine "<lang>"
        actionwords.iStartTheCoffeeMachine(lang);
        // Then message "<ready_message>" should be displayed
        actionwords.messageMessageShouldBeDisplayed(readyMessage);
    }

    @Test
    public void messagesAreBasedOnLanguageEnglishUid4575ba395fa046e69637570f4f935bc2() {
        messagesAreBasedOnLanguage("en", "Ready");
    }

    @Test
    public void messagesAreBasedOnLanguageFrenchUide2d71302d9064848b6e2af29c7e5117c() {
        messagesAreBasedOnLanguage("fr", "Pret");
    }
}

File: javascript/qunit/project_test.js
(function () {
  module('Coffee machine - Hiptest publisher sample', {
    setup: function () {
      this.actionwords = Object.create(Actionwords);
      this.actionwords.sut = CoffeeMachine();
    }
  });

  function simpleUse (lang, ready_message) {
    // Given I start the coffee machine "<lang>"
    this.actionwords.iStartTheCoffeeMachine(lang);
    // When I take a coffee
    this.actionwords.iTakeACoffee();
    // Then coffee should be served
    this.actionwords.coffeeShouldBeServed();
  }

  test('Simple use: English (uid:6617621c-6e93-4b66-9b48-cda962025bfc)', function () {
    simpleUse.apply(this, ['en', 'Ready']);
  });

  test('Simple use: French (uid:597940b5-9739-4f52-bc15-0a8f2d09921b)', function () {
    simpleUse.apply(this, ['fr', 'Pret']);
  });


  test('Full grounds does not block coffee (uid:393737bf-81f2-423e-a59c-de11deb6c80c)', function () {
    // Given the coffee machine is started
    this.actionwords.theCoffeeMachineIsStarted();
    // When I take "29" coffees
    this.actionwords.iTakeCoffeeNumberCoffees(29);
    // Then message "Ready" should be displayed
    this.actionwords.messageMessageShouldBeDisplayed("Ready");
    // When I take a coffee
    this.actionwords.iTakeACoffee();
    // Then coffee should be served
    this.actionwords.coffeeShouldBeServed();
    // And message "Empty grounds" should be displayed
    this.actionwords.messageMessageShouldBeDisplayed("Empty grounds");
    // When I fill the water tank
    this.actionwords.iFillTheWaterTank();
    // And I fill the beans tank
    this.actionwords.iFillTheBeansTank();
    // And I take "20" coffees
    this.actionwords.iTakeCoffeeNumberCoffees(20);
    // Then coffee should be served
    this.actionwords.coffeeShouldBeServed();
    // And message "Empty grounds" should be displayed
    this.actionwords.messageMessageShouldBeDisplayed("Empty grounds");
  });

  test('Water runs away (uid:3bbac854-fad6-444d-9a59-0faadbba2f55)', function () {
    // Given the coffee machine is started
    this.actionwords.theCoffeeMachineIsStarted();
    // When fifty coffees have been taken without filling the tank
    this.actionwords.fiftyCoffeesHaveBeenTakenWithoutFillingTheTank();
    // Then message "Fill tank" should be displayed
    this.actionwords.messageMessageShouldBeDisplayed("Fill tank");
    // When I take a coffee
    this.actionwords.iTakeACoffee();
    // Then coffee should be served
    this.actionwords.coffeeShouldBeServed();
    // When I take "10" coffees
    this.actionwords.iTakeCoffeeNumberCoffees(10);
    // Then coffee should not be served
    this.actionwords.coffeeShouldNotBeServed();
    // And message "Fill tank" should be displayed
    this.actionwords.messageMessageShouldBeDisplayed("Fill tank");
    // When I fill the water tank
    this.actionwords.iFillTheWaterTank();
    // Then message "Ready" should be displayed
    this.actionwords.messageMessageShouldBeDisplayed("Ready");
  });

  test('Beans run out (uid:c3df3d32-e883-496a-aeec-0e7cb9ca363a)', function () {
    // Given the coffee machine is started
    this.actionwords.theCoffeeMachineIsStarted();
    // When thirty eight coffees are taken without filling beans
    this.actionwords.thirtyEightCoffeesAreTakenWithoutFillingBeans();
    // Then coffee should be served
    this.actionwords.coffeeShouldBeServed();
    // And message "Fill beans" should be displayed
    this.actionwords.messageMessageShouldBeDisplayed("Fill beans");
    // When I take "2" coffees
    this.actionwords.iTakeCoffeeNumberCoffees(2);
    // Then coffee should be served
    this.actionwords.coffeeShouldBeServed();
    // And message "Fill beans" should be displayed
    this.actionwords.messageMessageShouldBeDisplayed("Fill beans");
    // When I take a coffee
    this.actionwords.iTakeACoffee();
    // Then coffee should not be served
    this.actionwords.coffeeShouldNotBeServed();
  });

  function messagesAreBasedOnLanguage (lang, ready_message) {
    // When I start the coffee machine "<lang>"
    this.actionwords.iStartTheCoffeeMachine(lang);
    // Then message "<ready_message>" should be displayed
    this.actionwords.messageMessageShouldBeDisplayed(ready_message);
  }

  test('Messages are based on language: English (uid:22759344-4f50-4f05-b0c7-bb6775b2bdb4)', function () {
    messagesAreBasedOnLanguage.apply(this, ['en', 'Ready']);
  });

  test('Messages are based on language: French (uid:c6049f15-75ee-43b1-97c4-5815a88324ce)', function () {
    messagesAreBasedOnLanguage.apply(this, ['fr', 'Pret']);
  });


  test('No messages are displayed when machine is shut down (uid:e9d93b85-30d5-4ce1-a85a-eb1b41992598)', function () {
    // Given the coffee machine is started
    this.actionwords.theCoffeeMachineIsStarted();
    // When I shutdown the coffee machine
    this.actionwords.iShutdownTheCoffeeMachine();
    // Then message "" should be displayed
    this.actionwords.messageMessageShouldBeDisplayed();
  });
})();
